### PR TITLE
Update FsCheck3 support to release as a pre-release version of Expecto.FsCheck

### PR DESCRIPTION
**Pull request by farlee2121**
This way we don't have to create a separate package. Should simplify adoption and trimming once FsCheck 3 reaches general release.

#450

### CODE REVIEW

**Comment by farlee2121** [comment]
@ratsclub Please let me know when you've merged your build system updates (or if you don't have time for it and want me to pick it up)

**Comment by JohnTheGr8** [comment]
relevant message from paket:

```
Paket failed with
-> There was a version conflict during package resolution.
     Conflict detected:
      - Dependencies file requested package FsCheck: >= 3.0 < 4.0
      - Available versions:
        - (3.0.0-beta2, [https://api.nuget.org/v3/index.json])
        - (3.0.0-beta1, [https://api.nuget.org/v3/index.json])
        - (3.0.0-alpha5, [https://api.nuget.org/v3/index.json])
        - (3.0.0-alpha4, [https://api.nuget.org/v3/index.json])
        - (3.0.0-alpha3, [https://api.nuget.org/v3/index.json])
        - (3.0.0-alpha2, [https://api.nuget.org/v3/index.json])
        - (3.0.0-alpha1, [https://api.nuget.org/v3/index.json])
        - (2.16.5, [https://api.nuget.org/v3/index.json])
        - (2.16.4, [https://api.nuget.org/v3/index.json])
        - (2.16.3, [https://api.nuget.org/v3/index.json])
        - (2.16.2, [https://api.nuget.org/v3/index.json])
        - (2.16.1, [https://api.nuget.org/v3/index.json])
        - (2.16.0, [https://api.nuget.org/v3/index.json])
        - (2.15.3, [https://api.nuget.org/v3/index.json])
        - (2.15.2, [https://api.nuget.org/v3/index.json])
        - (2.15.1, [https://api.nuget.org/v3/index.json])
        - (2.15.0, [https://api.nuget.org/v3/index.json])
        - (2.14.6, [https://api.nuget.org/v3/index.json])
        - (2.14.5, [https://api.nuget.org/v3/index.json])
        - (2.14.4, [https://api.nuget.org/v3/index.json])
        - (2.14.3, [https://api.nuget.org/v3/index.json])
        - (2.14.2, [https://api.nuget.org/v3/index.json])
        - (2.14.1, [https://api.nuget.org/v3/index.json])
        - (2.14.0, [https://api.nuget.org/v3/index.json])
        - (2.13.0, [https://api.nuget.org/v3/index.json])
        - (2.12.1, [https://api.nuget.org/v3/index.json])
        - (2.12.0, [https://api.nuget.org/v3/index.json])
        - (2.11.0, [https://api.nuget.org/v3/index.json])
        - (2.10.10, [https://api.nuget.org/v3/index.json])
        - (2.10.9, [https://api.nuget.org/v3/index.json])
        - (2.10.8, [https://api.nuget.org/v3/index.json])
        - (2.10.7, [https://api.nuget.org/v3/index.json])
        - (2.10.6, [https://api.nuget.org/v3/index.json])
        - (2.10.5, [https://api.nuget.org/v3/index.json])
        - (2.10.4, [https://api.nuget.org/v3/index.json])
        - (2.10.3, [https://api.nuget.org/v3/index.json])
        - (2.10.2, [https://api.nuget.org/v3/index.json])
        - (2.10.1, [https://api.nuget.org/v3/index.json])
        - (2.10.0, [https://api.nuget.org/v3/index.json])
        - (2.9.2, [https://api.nuget.org/v3/index.json])
        - (2.9.1, [https://api.nuget.org/v3/index.json])
        - (2.9.0, [https://api.nuget.org/v3/index.json])
        - (2.9.0-rc3, [https://api.nuget.org/v3/index.json])
        - (2.9.0-rc2, [https://api.nuget.org/v3/index.json])
        - (2.9.0-rc1, [https://api.nuget.org/v3/index.json])
        - (2.8.2, [https://api.nuget.org/v3/index.json])
        - (2.8.1, [https://api.nuget.org/v3/index.json])
        - (2.8.0, [https://api.nuget.org/v3/index.json])
        - (2.7.2, [https://api.nuget.org/v3/index.json])
        - (2.7.1, [https://api.nuget.org/v3/index.json])
        - (2.7.0, [https://api.nuget.org/v3/index.json])
        - (2.6.3, [https://api.nuget.org/v3/index.json])
        - (2.6.2, [https://api.nuget.org/v3/index.json])
        - (2.6.1, [https://api.nuget.org/v3/index.json])
        - (2.6.0, [https://api.nuget.org/v3/index.json])
        - (2.5.0, [https://api.nuget.org/v3/index.json])
        - (2.4.0, [https://api.nuget.org/v3/index.json])
        - (2.3.0, [https://api.nuget.org/v3/index.json])
        - (2.2.5, [https://api.nuget.org/v3/index.json])
        - (2.2.4, [https://api.nuget.org/v3/index.json])
        - (2.2.3, [https://api.nuget.org/v3/index.json])
        - (2.2.2, [https://api.nuget.org/v3/index.json])
        - (2.2.1, [https://api.nuget.org/v3/index.json])
        - (2.2.0, [https://api.nuget.org/v3/index.json])
        - (2.1.0, [https://api.nuget.org/v3/index.json])
        - (2.0.7, [https://api.nuget.org/v3/index.json])
        - (2.0.6, [https://api.nuget.org/v3/index.json])
        - (2.0.5, [https://api.nuget.org/v3/index.json])
        - (2.0.5-a1507312207, [https://api.nuget.org/v3/index.json])
        - (2.0.4, [https://api.nuget.org/v3/index.json])
        - (2.0.3, [https://api.nuget.org/v3/index.json])
        - (2.0.2, [https://api.nuget.org/v3/index.json])
        - (2.0.1, [https://api.nuget.org/v3/index.json])
        - (2.0.1-rc1, [https://api.nuget.org/v3/index.json])
        - (2.0.0, [https://api.nuget.org/v3/index.json])
        - (2.0.0-alpha, [https://api.nuget.org/v3/index.json])
        - (1.0.4, [https://api.nuget.org/v3/index.json])
        - (1.0.3, [https://api.nuget.org/v3/index.json])
        - (1.0.2, [https://api.nuget.org/v3/index.json])
        - (1.0.1, [https://api.nuget.org/v3/index.json])
        - (1.0.0, [https://api.nuget.org/v3/index.json])
        - (0.9.4, [https://api.nuget.org/v3/index.json])
        - (0.9.3, [https://api.nuget.org/v3/index.json])
        - (0.9.2, [https://api.nuget.org/v3/index.json])
        - (0.9.1, [https://api.nuget.org/v3/index.json])
        - (0.9.0.1, [https://api.nuget.org/v3/index.json])
        - (0.9.0, [https://api.nuget.org/v3/index.json])
        - (0.8.3, [https://api.nuget.org/v3/index.json])
        - (0.8.2, [https://api.nuget.org/v3/index.json])
        - (0.8.1, [https://api.nuget.org/v3/index.json])
        - (0.8.0, [https://api.nuget.org/v3/index.json])
        - (0.7.1, [https://api.nuget.org/v3/index.json])

     Please try to relax some conditions or resolve the conflict manually (see http://fsprojects.github.io/Paket/nuget-dependencies.html#Use-exactly-this-version-constraint).
```

File: paket.dependencies [hunk 1]
```diff
@@ -13,7 +13,7 @@ nuget BenchmarkDotNet ~> 0.13.5
 group FsCheck3
   source https://api.nuget.org/v3/index.json
 
-  nuget FsCheck ~> 3.0.0-beta2
+  nuget FsCheck ~> 3
```
[the remark this answers is not in the document]
Comment (farlee2121) [reply paket.dependencies:16]: Shoot, you're right. I got hasty. Thanks for the save. I'll get a fix out asap

[the remark this answers is not in the document]
Comment (JohnTheGr8) [reply paket.dependencies:16]: I could open a PR for #463 once you're done unless you've started work on it

[the remark this answers is not in the document]
Comment (farlee2121) [reply paket.dependencies:16]: Hmm. Maybe I do need some help here. The security issues are plaguing [this pr](https://github.com/haf/expecto/pull/465). I update the vulnerable packages to suggested versions, but it looks like WhiteSource keeps failing the check without re-evaluating. Clicking re-run just results in the same `The contents of this commit did not trigger a Mend scan.` message.

[the remark this answers is not in the document]
Comment (JohnTheGr8) [reply paket.dependencies:16]: > Hmm. Maybe I do need some help here. The security issues are plaguing [this pr](https://github.com/haf/expecto/pull/465). I update the vulnerable packages to suggested versions, but it looks like WhiteSource keeps failing the check without re-evaluating. Clicking re-run just results in the same `The contents of this commit did not trigger a Mend scan.` message.

I think the bot is stuck, your changes seem fine... mind if I open (yet another) PR to hopefully fix it?

[the remark this answers is not in the document]
Comment (farlee2121) [reply paket.dependencies:16]: I tried closing the PR and opening a new one earlier, but it didn't work. Maybe we need to completely start over with a new branch and new commits. I'll check real quick

[the remark this answers is not in the document]
Comment (JohnTheGr8) [reply paket.dependencies:16]: yup, i suspect that it skips the scan because the initial commit is the same in both PRs you opened...

**Pull request merged**